ethereum/executionlayer/src/test/java/tech/pegasys/teku/ethereum/executionlayer/client/Web3JExecutionEngineClientTest.java

```diff
@@ -496,6 +511,77 @@ void shouldTransitionConfigurationRoundtrip() {
+  void shouldRoundtripWithMockedWebServer() throws InterruptedException, IOException {
+
+    mockWebServer.start();
+
+    final Web3JExecutionEngineClient eeClient =
+        new Web3JExecutionEngineClient(
+            "http://localhost:" + mockWebServer.getPort(), timeProvider, Optional.empty());
+
+    final Bytes32 latestValidHash =
+        Bytes32.fromHexString("0x135bc3400c2839fd856a524871200bd5e362db615fc4565e1870ed9a2a936464");
+    final String validationError = "error";
+    final PayloadStatus payloadStatusResponse =
+        PayloadStatus.invalid(Optional.of(latestValidHash), Optional.of(validationError));
+
+    final String bodyResponse =
+        "{\"jsonrpc\": \"2.0\", \"id\": 0, \"result\":"
+            + "{\"payloadStatus\" : { \"status\": \"INVALID\", \"latestValidHash\": "
+            + latestValidHash
+            + ", \"validationError\": "
+            + validationError
+            + "}}}";
+
+    mockWebServer.enqueue(
+        new MockResponse()
+            .setResponseCode(200)
+            .setBody(bodyResponse)
+            .addHeader("Content-Type", "application/json"));
+
+    final ForkChoiceStateV1 forkChoiceStateV1Orig =
+        new ForkChoiceStateV1(
+            Bytes32.fromHexString(
+                "0x235bc3400c2839fd856a524871200bd5e362db615fc4565e1870ed9a2a936464"),
+            Bytes32.fromHexString(
+                "0x367cbd40ac7318427aadb97345a91fa2e965daf3158d7f1846f1306305f41bef"),
+            Bytes32.fromHexString(
+                "0xfd18cf40cc907a739be483f1ca0ee23ad65cdd3df23205eabc6d660a75d1f54e"));
+
+    final PayloadAttributesV1 payloadAttributesV1Request =
+        new PayloadAttributesV1(
+            UInt64.valueOf(10),
+            Bytes32.fromHexString(
+                "0x367cbd40ac7318427aadb97345a91fa2e965daf3158d7f1846f1306305f41bef"),
+            Bytes20.fromHexString("0xfd18cf40cc907a739be483f1ca0ee23ad65cdd3d"));
+
+    final SafeFuture<Response<ForkChoiceUpdatedResult>> futureResponseForkChoiceUpdatedResult =
+        eeClient.forkChoiceUpdated(forkChoiceStateV1Orig, Optional.of(payloadAttributesV1Request));
+
+    final RecordedRequest request = mockWebServer.takeRequest();
+
+    assertThat(request.getBody().readString(StandardCharsets.UTF_8))
+        .isEqualTo(
+            "{\"jsonrpc\":\"2.0\",\"method\":\"engine_forkchoiceUpdatedV1\",\"params\":["
+                + "{\"headBlockHash\":\"0x235bc3400c2839fd856a524871200bd5e362db615fc4565e1870ed9a2a936464\","
+                + "\"safeBlockHash\":\"0x367cbd40ac7318427aadb97345a91fa2e965daf3158d7f1846f1306305f41bef\","
+                + "\"finalizedBlockHash\":\"0xfd18cf40cc907a739be483f1ca0ee23ad65cdd3df23205eabc6d660a75d1f54e\"},"
+                + "{\"timestamp\":\"0xa\","
+                + "\"prevRandao\":\"0x367cbd40ac7318427aadb97345a91fa2e965daf3158d7f1846f1306305f41bef\","
+                + "\"suggestedFeeRecipient\":\"0xfd18cf40cc907a739be483f1ca0ee23ad65cdd3d\"}],"
+                + "\"id\":0}");
```

Comment:
I wonder if it would be better to parse the expected and actual into generic JsonNode and compare those so that the specific string formatting doesn't matter as long as it's semantically the same JSON.